File: Python/00-primero/01-script.py
#!/usr/bin/env python3

# Programa una funcion que te devuelva el texto recortado desgun el numero de caracteres
# indicado. miFuncion("Hola mundo",4) devolvera "Hola"


def recortar_texto(cadena: str, longitud: int) -> str:
    """Recorta el texto según el número de caracteres indicado.
    
    Args:
        cadena (str): La cadena de texto a recortar.
        longitud (int): El número de caracteres a mantener.
    
    Returns:
        str: La cadena recortada o un mensaje de error si los argumentos no son válidos.
    """
    if not isinstance(cadena, str) or not isinstance(longitud, int):
        return "Debes ingresar una cadena y una longitud válidas."
    
    if longitud < 0:
        return "La longitud debe ser un número entero positivo."

    return cadena[:longitud]

# Ejemplo de uso
cadena_recortada = recortar_texto("Hola mundo", 5)
print(cadena_recortada)

cadena_recortada = recortar_texto(87,5)
print(cadena_recortada)

File: Php/00-primero/01-script.php
<?php
/**
 * Recorta el texto según el número de caracteres indicado.
 *
 * @param string $cadena La cadena de texto a recortar.
 * @param int $longitud El número de caracteres a mantener.
 * @return string La cadena recortada o un mensaje de error si los argumentos no son válidos.
 */

function recortar_texto($cadena, $longitud) {
    // Validar que la cadena no esté vacía y sea una cadena de texto
    if (!is_string($cadena) || empty($cadena)) {
        return "Debes ingresar una cadena válida.";
    }

    // Validar que longitud sea un número entero positivo
    if (!is_int($longitud) || $longitud < 0) {
        return "Debes ingresar una longitud válida.";
    }

    // Recortar el texto
    return substr($cadena, 0, $longitud);
}

// Ejemplo de uso
$cadena_recortada = recortar_texto("Hola mundo", 5);
echo $cadena_recortada . PHP_EOL;
?>